src/Workspaces/Core/Portable/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.BasicBlockAnalysisData.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis.SymbolUsageAnalysis
+{
+    internal static partial class SymbolUsageAnalysis
+    {
+        /// <summary>
+        /// Analysis data for a particular <see cref="BasicBlock"/> for <see cref="ControlFlowGraph"/>
+        /// based dataflow analysis OR for the entire executable code block for high level operation
+        /// tree based analysis.
+        /// </summary>
+        private sealed class BasicBlockAnalysisData : IEquatable<BasicBlockAnalysisData>
+        {
+            private static readonly ObjectPool<BasicBlockAnalysisData> s_pool =
+                new ObjectPool<BasicBlockAnalysisData>(() => new BasicBlockAnalysisData());
+
+            /// <summary>
+            /// Map from each symbol to possible set of reachable write operations that are live at current program point.
+            /// A write is live if there is no intermediate write operation that overwrites it.
+            /// </summary>
+            private readonly Dictionary<ISymbol, PooledHashSet<IOperation>> _reachingWrites;
```

Comment:
>_reachingWrites [](start = 76, length = 15)

This is the primary data being tracked for each basic block in our custom analysis. We track all the symbol write operations that are live at the start of the basic block corresponding to this data.